dubbo-common/src/main/java/org/apache/dubbo/common/utils/StringUtils.java

```diff
@@ -478,10 +477,23 @@ public static boolean isContains(String[] values, String value) {
         return false;
     }
 
-    public static boolean isNumeric(String str) {
-        if (str == null) {
+    public static boolean isNumeric(String str, boolean allowDot) {
+        if (str == null || str.isEmpty()) {
             return false;
         }
+        if (allowDot) {

```

Comment:
Hi,

I am not a fan of recursive call, especially for such a small function. I think this can be implemented by at most 2 for loops.

Can you replace it by a non-recursive call?

Reply:
I have modified it.

Reply:
Hi, 

How about this:

```java
public static boolean isNumeric(String str, boolean allowDot) {
        if (str == null || str.isEmpty()) {
            return false;
        }
        int dotCount = 0;
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) == '.') {
                if (allowDot && dotCount == 0) {
                    dotCount++;
                    continue;
                } else {
                    return false;
                }
            }
            if (!Character.isDigit(str.charAt(i))) {
                return false;
            }
        }
        return true;
    }
```

This only need 1 loop. And also support something like "1." as a numeric.

```
jshell> Double.valueOf("1.")
$1 ==> 1.0
```

I also add some more unit test

```java
assertThat(StringUtils.isNumeric("123.3.3", true), is(false));
assertThat(StringUtils.isNumeric("123.", true), is(true));
```

Reply:
So is there a support for “.1”?

Reply:
Definitely yes. You can add it to the UT.

Reply:
do it.